### PR TITLE
Update security.md

### PR DESCRIPTION
**Pull request by yaron2**
Updates with mTLS details.

### CODE REVIEW

File: concepts/security/security.md [hunk 1]
```diff
@@ -6,13 +6,24 @@ Dapr sidecar runs close to the application through **localhost**. Dapr assumes i
...
+When root certs are replaced (Secret in Kubernetes mode and filesystem for self hosted mode), Sentry will pick them up and re-build the trust bundle without needing to restart, again with zero downtime to Sentry.
+
+When a new Dapr sidecar inits, it will first check if mTLS is enabled. if it is, an ECDSA private key and certificate signing request are generated and sent to Sentry via a gRPC interface. The communication between the Dapr sidecar and Sentry is authenticated using the trust chain cert, which is injected to each Dapr sidecar by the Dapr sidecar injector.
+
+The secret that hold the root certificates are scoped to the namespace in which the Dapr components are deployed to and are only accessible by the Dapr system pods.
+
+Dapr also support strong identities when deployed on Kubernetes, relying on a pod's Service Account token which is sent as part of the CSR request to Sentry.
+
+Mutual TLS can be turned off by editing the default configuration that is deployed with Dapr via the `spec.mtls.enabled` field.
+
```
Comment (youngbupark) [inline concepts/security/security.md]: maybe it is good to provide exact yaml

**Comment by yaron2** [comment]
Not here, there's another PR coming after that.

**Comment by youngbupark** [comment]
> Not here, there's another PR coming after that.

ok then. it would be good to have a link after that

File: concepts/security/security.md (hunk 1 above)
Comment (yaron2) [reply concepts/security/security.md]: Now it's in here..

**Comment by yaron2** [comment]
@msfussell since this is a concept doc, I removed the instructions and instead linked to a placeholder which will contain the next PR for the actual instructions in the How-To section.

This is in order to clear ambiguity and prevent duplicate content, leaving the concepts for explaining the concepts and free of implementation details.

**Comment by msfussell** [comment]
It would be good to have a diagram in the concepts to iwth words on how the sentry service works, in the same way that the placement service was done for Actors here https://github.com/dapr/docs/blob/master/concepts/actor/actor_overview.md. This prevents us from having to repeat whet the sentry service does and how it works.

**Comment by yaron2** [comment]
> It would be good to have a diagram in the concepts to iwth words on how the sentry service works, in the same way that the placement service was done for Actors here https://github.com/dapr/docs/blob/master/concepts/actor/actor_overview.md. This prevents us from having to repeat whet the sentry service does and how it works.

When you say "with words", you mean the existing description, yes? so only the diagram.

**Comment by msfussell** [comment]
If asked "how does the dapr sentry service work" sending a link to this concept file would satisfy that ask including saying words along the line of "each dapr instance get a cert generated from the sentry service" etc. It is a significant piece of functionality and there will be many ops people who will want to understand how this works to satisfy security requirements in their organization.

**Comment by yaron2** [comment]
> If asked "how does the dapr sentry service work" sending a link to this concept file would satisfy that ask including saying words along the line of "each dapr instance get a cert generated from the sentry service" etc. It is a significant piece of functionality and there will be many ops people who will want to understand how this works to satisfy security requirements in their organization.

The diagram is needed as well. The words that say that each Dapr instance is getting it's own cert are already there.

**Pull request merged**